data/storeDEPRECATED/mongo/mongo_bucket_store_client.go

```diff
@@ -49,19 +49,22 @@ func NewMongoBucketStoreClient(config *goComMgo.Config, logger *log.Logger) (*Mo
 
 /* bucket methods */
 
-// Look for a single bucket based on its Id
-func (c *MongoBucketStoreClient) Find(ctx context.Context, bucket schema.IBucket) (result *schema.IBucket, err error) {
+// Look for a single bucket based on its Id in cold and daily
+func (c *MongoBucketStoreClient) Find(ctx context.Context, bucket schema.IBucket, dataType string) (result *schema.IBucket, err error) {
 
 	if bucket.GetId() != "" {
 		var query bson.M = bson.M{}
 		query["_id"] = bucket.GetId()
 		opts := options.FindOne()
 		opts.SetSort(bson.D{primitive.E{Key: "_id", Value: -1}})
-		if err = c.Collection("hotDailyCbg").FindOne(ctx, query, opts).Decode(&result); err != nil && err != mongo.ErrNoDocuments {
-			c.log.WithError(err)
-			return result, err
-		}
 
+		for _, collectionPrefix := range dailyPrefixCollections {
+			collectionName := collectionPrefix + dataType
+			if err = c.Collection(collectionName).FindOne(ctx, query, opts).Decode(&result); err != nil && err != mongo.ErrNoDocuments {
+				c.log.WithError(err)
+				return result, err
+			}
+		}
```

Comment:
If data is present in both collection we will do two mongo requests and always return the result from last dailyPrefixCollections.
```suggestion
		for _, collectionPrefix := range dailyPrefixCollections {
			collectionName := collectionPrefix + dataType
			if err = c.Collection(collectionName).FindOne(ctx, query, opts).Decode(&result); err != nil && err != mongo.ErrNoDocuments {
				c.log.WithError(err)
				return result, err
			} else if err != mongo.ErrNoDocuments {
			       return result, nil
			}
		}
```